### PR TITLE
feat: add btp network policy

### PR DESCRIPTION
**Pull request by ofirmad**
# Pull Request Template

## Prerequisites

* If there is no issue related to your pull request - open one and assign yourself to it. If you're proposing a solution to an already opened issue - simply assign yourself to it.

## Motivation

*Describe the problem you're trying to solve or the feature you're proposing in addition to a link to the related issue.*

## Approach

_How does this change address the problem?_

## Pull Request status

* [ ] Initial implementation
* [ ] Refactoring
* [ ] Unit tests
* [ ] Integration tests
* [x] contribution

Feel free to construct the checklist with whatever items seem most reasonable for your change. You could disassemble the Implementation part to even smaller separate checklist items if you're working on something big for example. But do make the effort to provide a checklist of some sort so that the core team as well as the community can have an idea about the progress of your Pull Request.

## Third-party code

If you use third party code with your contribution such as, components, libraries, or snippets make sure to mention that.

## Work in Progress label

For as long as development of your Pull Request is still ongoing you MUST label it with a **wip** label as well as prefix the name of the PR with *[WIP]*.

For example: *[WIP] Service brokers API*

### CODE REVIEW

File: sapbtp-operator-charts/templates/networkpolicy.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        port: 9443      # webhook service
+      - protocol: TCP
+        port: 8081     # readiness and liveness probe
+    - from:
+      - podSelector:
+          matchLabels:
+            app.kubernetes.io/name: prometheus
+      ports:
+      - protocol: TCP
+        port: 8443      # prometheus metrics
```
Comment (gadisn) [inline sapbtp-operator-charts/templates/networkpolicy.yaml:33]: 8443 is the service port, which might also be used by prometheus.
But I don't think that this chart has a servicemonitor CR, so prometheus isn't used here

Comment (ofirmad) [reply sapbtp-operator-charts/templates/networkpolicy.yaml:33]: I will remove prometheus condition

**Comment by ofirmad** [comment]
This PR is about adding netowrkpolicy to btp operator.
It is needed for security, so that btp will have connection only to it needs to connect with/needs to connect with it.
I added it with enables: false so it won't affect you unless to would like to use it.

**Review by gadisn — COMMENTED** [review]
Looks good to me

File: sapbtp-operator-charts/templates/networkpolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  name: btp-networkpolicy
+  namespace: {{ .Release.Namespace }}
+spec:
+  podSelector: {}
+  egress:
+    - ports:
+      - protocol: TCP
+        port: 443
+      - protocol: TCP
+        port: 53           # dns
```
[the remark this answers is not in the document]
Comment (ofirmad) [reply sapbtp-operator-charts/templates/networkpolicy.yaml]: I checked the changes with helm template and it works

[the remark this answers is not in the document]
Comment (kerenlahav) [reply sapbtp-operator-charts/templates/networkpolicy.yaml]: I don't see these ports are being used in the btp operator, is this something specific to your deployment?

[the remark this answers is not in the document]
Comment (ofirmad) [reply sapbtp-operator-charts/templates/networkpolicy.yaml]: There are no ports at all now, I did it generic.
We use 53 and 8053 ports to talk to the DNS server in our requests to convert the address to an IP

File: sapbtp-operator-charts/values.yaml [hunk 3]
```diff
@@ -93,3 +93,6 @@ externalImages:
       repository: bitnami/kubectl
       sha: ""
       tag: latest
+
+networkPolicy:
```
Comment (kerenlahav) [inline sapbtp-operator-charts/values.yaml:97]: please set a comment with an example for a valid ingress and egress values like [here](https://github.com/SAP/sap-btp-service-operator/blob/main/sapbtp-operator-charts/values.yaml#L55)

Comment (ofirmad) [reply sapbtp-operator-charts/values.yaml:97]: Done

**Pull request closed without merging**